File: seed/csharp-sdk/trace/src/SeedTrace/Playlist/Types/Playlist.cs
using System.Text.Json.Serialization;
using SeedTrace.Core;

namespace SeedTrace;

public record Playlist
{
    [JsonPropertyName("playlist_id")]
    public required string PlaylistId { get; set; }

    [JsonPropertyName("owner-id")]
    public required string OwnerId { get; set; }

    [JsonPropertyName("name")]
    public required string Name { get; set; }

    [JsonPropertyName("problems")]
    public IEnumerable<string> Problems { get; set; } = new List<string>();

    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: seed/php-model/trace/src/Playlist/Playlist.php
<?php

namespace Seed\Playlist;

use Seed\Core\Json\JsonSerializableType;
use Seed\Playlist\Traits\PlaylistCreateRequest;
use Seed\Core\Json\JsonProperty;

class Playlist extends JsonSerializableType
{
    use PlaylistCreateRequest;

    /**
     * @var string $playlistId
     */
    #[JsonProperty('playlist_id')]
    public string $playlistId;

    /**
     * @var string $ownerId
     */
    #[JsonProperty('owner-id')]
    public string $ownerId;

    /**
     * @param array{
     *   name: string,
     *   problems: array<string>,
     *   playlistId: string,
     *   ownerId: string,
     * } $values
     */
    public function __construct(
        array $values,
    )
    {
        $this->name = $values['name'];$this->problems = $values['problems'];$this->playlistId = $values['playlistId'];$this->ownerId = $values['ownerId'];
    }

    /**
     * @return string
     */
    public function __toString(): string {
        return $this->toJson();
    }
}
